internal/b/blueprints.go

```diff
@@ -68,11 +75,22 @@ type Room struct {
 	Events     []Event
 }
 
+type ApplicationService struct {
+	ID              string
+	HSToken         string
+	ASToken         string
+	URL             string
+	SenderLocalpart string
+	RateLimited     bool
+}
+
 type Event struct {
-	Type     string
-	Sender   string
-	StateKey *string
-	Content  map[string]interface{}
+	Type           string
+	Sender         string
+	OriginServerTS uint64
+	StateKey       *string
+	PrevEvents     []string
+	Content        map[string]interface{}
```

Comment:
Please do not add these fields.

Reply:
TODO: These are used in https://github.com/matrix-org/complement/pull/68